tests/test_2591_nan_to_num.py

```diff
@@ -11,5 +11,6 @@
 
 def test():
     ak.nan_to_num(
-        ak.Array([1, 2, 3], backend="jax"), nan=ak.Array([1, 2, 3], backend="jax")
+        ak.Array([1.1, 2.2, 3.3], backend="jax"),
+        nan=ak.Array([1.1, 2.2, 3.3], backend="jax"),
     )
```

Comment:
This change had nothing to do with the PR. A new version of JAX complains that integers are not an inexact type, so I made them floats.